sn_interface/src/network_knowledge/prefix_map/mod.rs

```diff
@@ -119,74 +128,38 @@ impl NetworkPrefixMap {
         self.closest(name, exclude).or_else(|| {
             self.sections
                 .iter()
-                .filter(|e| e.key().matches(&name.with_bit(0, !name.bit(0))))
-                .max_by_key(|e| e.key().bit_count())
-                .map(|entry| entry.value().clone())
+                .filter(|&(prefix, _)| prefix.matches(&name.with_bit(0, !name.bit(0))))
+                .max_by_key(|&(prefix, _)| prefix.bit_count())
+                .map(|(_, sap)| sap.clone())
         })
     }
 
     /// Returns all known sections SAP.
     pub fn all(&self) -> Vec<SectionAuthorityProvider> {
         self.sections
             .iter()
-            .map(|e| e.value().value.clone())
+            .map(|(_, sap)| sap.value.clone())
             .collect()
     }
 
     /// Get `SectionAuthorityProvider` of a known section with the given prefix.
     #[allow(unused)]
     pub fn get(&self, prefix: &Prefix) -> Option<SectionAuthorityProvider> {
-        self.sections
-            .get(prefix)
-            .map(|entry| entry.value().value.clone())
+        self.sections.get(prefix).map(|sap| sap.value.clone())
```

Comment:
Use your helpers when available
```suggestion
        self.get_signed(prefix).map(|sap| sap.value)
```

Reply:
Aye! Will keep that in mind :+1: